waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/DatabaseMappingImpl.java

```diff
@@ -77,14 +83,28 @@ public DatabaseMappingImpl(MetaStoreMapping metaStoreMapping, QueryMapping query
   public Table transformOutboundTable(Table table) {
     table.setDbName(metaStoreMapping.transformOutboundDatabaseName(table.getDbName()));
     if (table.isSetViewExpandedText()) {
-      table.setViewExpandedText(
-          queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewExpandedText()));
+      try {
+        log.debug("Transforming ViewExpandedText: {}", table.getViewExpandedText());
+        table.setViewExpandedText(
+            queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewExpandedText()));
+      } catch (WaggleDanceException e) {
+        log.debug("Error while transforming databaseName in ViewExpandedText, keeping original", e);
+      }
     }
     if (table.isSetViewOriginalText()) {
-      table.setViewOriginalText(
-          queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewOriginalText()));
+      try {
+        log.debug("Transforming ViewOriginalTex: {}", table.getViewOriginalText());
+        table.setViewOriginalText(
+            queryMapping.transformOutboundDatabaseName(metaStoreMapping, table.getViewOriginalText()));
+      } catch (WaggleDanceException e) {
+        log.debug("Error while transforming databaseName in ViewOriginalTex, using ViewExpandedText if available", e);
+        if (table.isSetViewExpandedText()) {
+          table.setViewOriginalText(table.getViewExpandedText());
```

Comment:
Why do we do this instead of just leaving the original text as it was?

Reply:
I've discussed this with @teabot, since the original text is not parseable and the db's in it are wrong we felt it might be better to add the parseable and correctly transformed expanded view. Hard to say what is better, depends on how this value is used. I'm happy to keep original as well.